adapter/PreciceInterface.h

```diff
@@ -216,6 +222,24 @@ void Precice_ReadIterationCheckpoint(SimulationData *sim, double *v);
  */
 void Precice_WriteIterationCheckpoint(SimulationData *sim, double *v);
 
+/**
+ * @brief Reads iteration checkpoint (in dyna_precice)
```

Comment:
We need a comment for the motivation for the duplication here (and in the writing method). What is special in modal simulations?

Reply:
Saved data is different. The saved data in modal simulations doesnt exist in regular ones.

But an if(isModal) inside would work too. Except that the number of arguments changes too, so I'm not sure

Reply:
Great motivations, let's add it to a comment!